i18n.js

```diff
@@ -804,9 +783,60 @@ module.exports = (function() {
   };
 
   /**
-   * searches for locale in given object
+   * Setup fallbacks for a given language.
+   * @param {Array} acceptedLanguages  languages from the HTTP Accept-Language header
+   * @param {JSON} fallbacks           fallbacks option
+   * @param {String} lang              current language
+   * @param {String} parentLang        current parent language
+   * @param {JSON} wildCardChecked     flags check if fallback for current language has been set up
    */
+  var setupLanguageFallbacks = function(acceptedLanguages, fallbacks, lang, parentLang,
+    wildCardChecked) {
+    var fallback;
+
+    if (fallbacks) {
+      var acceptedLanguageIndex,
+        fallbackIndex;
+      // Check if we have a configured fallback set for this language
+      if (fallbacks[lang]) {
+        fallback = fallbacks[lang];
+        // Fallbacks for languages should be inserted
+        // where the original, unsupported language existed.
+        acceptedLanguageIndex = acceptedLanguages.indexOf(lang);
+        fallbackIndex = acceptedLanguages.indexOf(fallback);
+        if (fallbackIndex > -1) {
+          acceptedLanguages.splice(fallbackIndex, 1);
+        }
+        acceptedLanguages.splice(acceptedLanguageIndex + 1, 0, fallback);
+      }
+      // Check if we have a configured fallback
+      // for other territory variation of this parent language by using wildcards
+      else if (!wildCardChecked[parentLang] && fallbacks[parentLang + '-*']) {
+        fallback = fallbacks[parentLang + '-*'];
+        acceptedLanguageIndex = acceptedLanguages.indexOf(lang);
+        fallbackIndex = acceptedLanguages.indexOf(fallback);
+        if (fallbackIndex > -1) {
+          acceptedLanguages.splice(fallbackIndex, 1);
+        }
+        acceptedLanguages.splice(acceptedLanguageIndex + 1, 0, fallback);
+        wildCardChecked[parentLang] = true;
+      }
+      // Check if we have a configured fallback set for the parent language of the locale.
+      if (fallbacks[parentLang]) {
+        fallback = fallbacks[parentLang];
+        // Fallbacks for a parent language should be inserted
+        // to the end of the list, so they're only picked
+        // if there is no better match.
+        if (acceptedLanguages.indexOf(fallback) < 0) {
+          acceptedLanguages.push(fallback);
```

Comment:
this function should not modify the parameters (AKA making side effect), it should return something